.github/workflows/actions/publish-gh-pages/action.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish GitHub Pages
+description: Downloads compiled code from the artifact cache and deploys it to GitHub Pages
+
+inputs:
+  github_token:
+    required: true
+  pr-number:
+    required: false
+    type: number
+  repository:
+    default: 'solana-web3.js'
+    type: string
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/download-artifact@v4
+      with:
+        path: .ghpages-deploy
+        pattern: ghpages-deploy-artifacts*
+        merge-multiple: true
+
+    - name: Deploy to Github Pages
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        destination_dir: ${{ inputs.pr-number }}/
```

Comment:
In the preview scenario, where there is a `destination_dir` set, I'm not sure if I need to set `keep_files: true` or not (so that multiple PRs can participate in the same `gh-pages` branch).